[sigmoid] memory planner C10 deps

dolpm opened the pull request:
Summary: perf-sensitive util functions for use in our memory planner

Test Plan: CI

Differential Revision: D73002726

Skylion007 commented on c10/util/UniqueVoidPtr.h:
```diff
@@ -74,6 +87,7 @@ class UniqueVoidPtr {
     if (get_deleter() != expected_deleter)
       return false;
+    has_deleter_ = new_deleter;
     ctx_ = std::unique_ptr<void, DeleterFnPtr>(ctx_.release(), new_deleter);
```
new_deleter can have std moved called on it here, right?

malfet commented on c10/core/impl/alloc_cpu.cpp:
```diff
@@ -64,17 +64,11 @@ static inline bool is_thp_alloc_enabled() {
-  static const auto pagesize = sysconf(_SC_PAGESIZE);
-  // for kernels that don't provide page size, default it to 4K
-  const size_t thp_alignment = (pagesize < 0 ? gPagesize : pagesize);
-  return (is_thp_alloc_enabled() ? thp_alignment : gAlignment);
```
Yes, looks this makes sense

malfet commented on c10/util/UniqueVoidPtr.h:
```diff
@@ -43,13 +43,16 @@ class UniqueVoidPtr {
   // Lifetime tied to ctx_
   void* data_;
   std::unique_ptr<void, DeleterFnPtr> ctx_;
+  bool has_deleter_{false};
```
Why do you need this instead of `ctx_. get_deleter() != nullptr`?

dolpm committed:
[sigmoid] memory planner C10 deps (#151275)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/151275

perf-sensitive util functions for use in our memory planner

Test Plan: CI

Reviewed By: georgiaphillips

Differential Revision: D73002726
